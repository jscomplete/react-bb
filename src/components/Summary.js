import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class Summary extends React.Component {
  static propTypes = {
    numberOfDeals: PropTypes.number.isRequired,
  };

  render() {
    return <div>Total number of deals: {this.props.numberOfDeals}</div>;
  }
}

export default connect((state) => ({
  numberOfDeals: state.deals.length,
}))(Summary);
