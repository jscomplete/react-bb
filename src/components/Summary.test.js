// 1. Runner (describe, it, test) ... Mochua
// 2. Assertion library (.equal, .deepEqual) ... Chai
// 3. Mocks (Stubs, doubles, ) ... Sinon

// JEST (JSX, modern JavaScript)
// Snapshot testing
// Shallow rendering

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Summary } from './Summary';

describe('Summary', () => {
  it('works', () => {
    const tree = TestRenderer.create(<Summary numberOfDeals={42} />).toJSON();

    expect(tree.children[1]).toEqual('42');

    expect(tree).toMatchSnapshot();
  });
});
