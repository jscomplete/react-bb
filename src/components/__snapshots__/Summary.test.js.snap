// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary works 1`] = `
<div>
  Total number of deals: 
  42
</div>
`;
